transform/coroutines.go

```diff
@@ -451,46 +500,48 @@ func (c *coroutineLoweringPass) analyzeFuncReturns(fn *asyncFunc) {
 
 // returnAnalysisPass runs an analysis pass which classifies the returns of all async functions.
 func (c *coroutineLoweringPass) returnAnalysisPass() {
-	for _, async := range c.asyncFuncs {
+	for _, async := range c.asyncFuncsOrdered {
 		c.analyzeFuncReturns(async)
 	}
 }
 
 // categorizeCalls categorizes all asynchronous calls into regular vs. async and matches them to their callers.
 func (c *coroutineLoweringPass) categorizeCalls() {
 	// Sort calls into their respective callers.
-	for _, async := range c.asyncFuncs {
-		async.calls = map[llvm.Value]struct{}{}
+	for _, async := range c.asyncFuncsOrdered {
+		async.calls = []llvm.Value{}
```

Comment:
You usually don't need to explicitly create a slice this way. Most operations are the same between nil slices and zero-length slices, except for comparing against nil. I believe that is the only difference.
(Maps are different, adding an element to a map requires the map to be created first).

Reply:
Fixed.